packages/api/src/tasks/checkAlerts.ts

```diff
@@ -160,27 +160,144 @@ export const notifyChannel = async ({
             }),
       });
 
-      // ONLY SUPPORTS SLACK WEBHOOKS FOR NOW
       if (webhook?.service === 'slack') {
-        await slack.postMessageToWebhook(webhook.url, {
-          text: message.title,
-          blocks: [
-            {
-              type: 'section',
-              text: {
-                type: 'mrkdwn',
-                text: `*<${message.hdxLink} | ${message.title}>*\n${message.body}`,
-              },
-            },
-          ],
-        });
+        await handleSendSlackWebhook(webhook, message);
+      } else if (webhook?.service === 'generic') {
+        await handleSendGenericWebhook(webhook, message);
       }
       break;
     }
     default:
       throw new Error(`Unsupported channel type: ${channel}`);
   }
 };
+
+const handleSendSlackWebhook = async (
+  webhook: IWebhook,
+  message: {
+    hdxLink: string;
+    title: string;
+    body: string;
+  },
+) => {
+  if (!webhook.url) {
+    throw new Error('Webhook URL is not set');
+  }
+
+  await slack.postMessageToWebhook(webhook.url, {
+    text: message.title,
+    blocks: [
+      {
+        type: 'section',
+        text: {
+          type: 'mrkdwn',
+          text: `*<${message.hdxLink} | ${message.title}>*\n${message.body}`,
+        },
+      },
+    ],
+  });
+};
+
+export const handleSendGenericWebhook = async (
+  webhook: IWebhook,
+  message: {
+    hdxLink: string;
+    title: string;
+    body: string;
+  },
+) => {
+  // QUERY PARAMS
+
+  if (!webhook.url) {
+    throw new Error('Webhook URL is not set');
+  }
+
+  let url: string;
+  // user input of queryParams is disabled on the frontend for now
+  if (webhook.queryParams) {
+    const queryParamsString = new URLSearchParams(
+      webhook.queryParams,
+    ).toString();
+
+    // user may have included params in both the url and the query params
+    // so they should be merged
+    const tmpURL = new URL(webhook.url);
+    if (queryParamsString) {
+      tmpURL.search = tmpURL.search
+        ? `${tmpURL.search}&${queryParamsString}`
+        : queryParamsString;
+    }
+    url = tmpURL.toString();
+  } else {
+    // if there are no query params given, just use the url
+    url = webhook.url;
+  }
+
+  // HEADERS
+  // TODO: handle real webhook security and signage after v0
+  // X-HyperDX-Signature FROM PRIVATE SHA-256 HMAC, time based nonces, caching functionality etc
+
+  const headers = {
+    'Content-Type': 'application/json', // default, will be overwritten if user has set otherwise
+    ...(webhook.headers?.toObject() || {}),
+  };
+
+  // BODY
+  let parsedBody: Record<string, string | number | symbol> = {};
+  if (webhook.body) {
+    const injectedBody = injectIntoPlaceholders(JSON.stringify(webhook.body), {
+      $HDX_ALERT_URL: message.hdxLink,
+      $HDX_ALERT_TITLE: message.title,
+      $HDX_ALERT_BODY: message.body,
+    });
+    parsedBody = JSON.parse(injectedBody);
```

Comment:
nit: trying to understand why we stringify and parse the body field instead of just iterating through values

Reply:
this is another artifact of the switch to the mongoose map - before ti was just a big json string - you can do it either way tbh this just regexes the entire string and replaces the values but now that its an object the injection can easily be mapped and happen on the individual value level

Reply:
this adds extra string->parse->string steps though so yeah, not ideal, ill adjust it

Reply:
leaving this to later, some annoyances around changing it and its "ok" for now...

Reply:
got curious this morning so tried it both ways, I found it more efficient to stick with injection happening on a full json string regex than going the object route, especially when you have to consider handling recursive deeply nested objects which could occur in many more complex body templating requests. The json parsing of the injected strings also has to happen regardless